core/controllers/pages_test.py

```diff
@@ -15,14 +15,149 @@
+
+
+class DonatePageTest(test_utils.GenericTestBase):
+    """Test for donate page."""
+
+    def test_loading_donate_page_loads_correctly(self):
+        response = self.testapp.get('/donate')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class ThanksPageTest(test_utils.GenericTestBase):
+    """Test for thanks page."""
+
+    def test_loading_thanks_page_loads_correctly(self):
+        response = self.testapp.get('/thanks')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class ForumPageTest(test_utils.GenericTestBase):
+    """Test for forum page."""
+
+    def test_loading_forum_page_loads_correctly(self):
+        response = self.testapp.get('/forum')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class TermsPageTest(test_utils.GenericTestBase):
+    """Test for terms page."""
+
+    def test_loading_terms_page_loads_correctly(self):
+        response = self.testapp.get('/terms')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+
+class PrivacyPageTest(test_utils.GenericTestBase):
+    """Test for privacy page."""
+
+    def test_loading_privacy_page_loads_correctly(self):
+        response = self.testapp.get('/privacy')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class AboutRedirectPageTest(test_utils.GenericTestBase):
+    """Test for about redirect page."""
+
+    def test_redirect_to_about_page_loads_correctly(self):
+        response = self.testapp.get('/credits')
+        self.assertEqual(response.status_int, 302)
+
+
+class FoundationRedirectPageTest(test_utils.GenericTestBase):
+    """Test for foundation redirect page."""
+
+    def test_redirect_to_foundation_page_loads_correctly(self):
+        response = self.testapp.get('/foundation')
+        self.assertEqual(response.status_int, 302)
+
+
+class TeachRedirectPageTest(test_utils.GenericTestBase):
+    """Test for teach redirect page."""
+
+    def test_redirect_to_teach_page_loads_correctly(self):
+        response = self.testapp.get('/participate')
+        self.assertEqual(response.status_int, 302)
+
+
+class ConsoleErrorPageTest(test_utils.GenericTestBase):
+    """Test for console error page."""
+
+    def test_loading_console_error_page_loads_correctly(self):
+        response = self.testapp.get('/console_errors')
+        self.assertEqual(response.status_int, 200)
+
+
+class MaintenancePageTest(test_utils.GenericTestBase):
+    """Test for maintenance page."""
+
+    def test_loading_maintenance_page_loads_correctly(self):
+        with self.swap(feconf, 'ENABLE_MAINTENANCE_MODE', True):
+            response = self.testapp.get('/admin')
+            self.assertEqual(response.status_int, 302)
```

Comment:
Check the redirect URL.

Reply:
What I mean is, 302 is a status code that means "we are redirecting the URL". Search "status code 302" for more details.

We should check that the URL redirects to the correct place. First you'll need to figure out what that place is; you can test that by going to /admin on your local dev server (with maintenance mode set to True in this case) and seeing what page it redirects to. Then, in this test, you will need to introspect the response to figure out what URL it is redirecting to (look up the webtest docs for more details) and assert that that URL is the expected one.

Hope this helps!

Reply:
Done!...

Reply:
def test_redirect_to_404_page_loads_correctly(self):
        with self.swap(feconf, 'ENABLE_MAINTENANCE_MODE', False):
            response = self.testapp.get('/admin')
            self.assertEqual(response.status_int, 302)

could you check here, even i was maintenance mode to false, then it gives status_int 302.

Reply:
@netajik I'm unable to understand your problem, why is the test name `test_redirect_to_404_page_loads_correctly` needs maintainence_mode?

Reply:
Here i want to test maintenance page when the ENABLE_MAINTENANCE_MODE is False. So then the page not redirect to url ="/admin" page. so it gives error. But in below code it redirect ot admin page.

def test_redirect_to_404_page_loads_correctly(self):
with self.swap(feconf, 'ENABLE_MAINTENANCE_MODE', False):
response = self.testapp.get('/admin')
self.assertEqual(response.status_int, 302).

Reply:
Ah I see, I think that's because you are trying to check the admin page as a guest user then you'll get redirected to the login page (I'm not sure!). Also, try having a meaningful test name (See the [3rd bullet of this wiki page](https://github.com/oppia/oppia/wiki/Writing-backend-tests#some-general-guidelines-for-writing-good-tests) for more info).

Reply:
Hello @seanlip, would you review my PR.

Reply:
@DubeySandeep is the primary reviewer for this PR; please ask him for reviews going forward (but only after you have fully resolved his comments). Thanks!

Reply:
Hello @DubeySandeep, i was run test setting  ENABLE_MAINTENANCE_MODE false. 


**def test_redirect_splash_page_loads_correctly(self):
        with self.swap(feconf, 'ENABLE_MAINTENANCE_MODE', False):
            response = self.testapp.get('/admin')
            self.assertEqual(response.status_int, 302)
            self.assertIn('splash', response.headers['location'])**

Then i am getting below error. So how can i solve it. Could you please tell me once.

**AssertionError: 'splash' not found in 'https://www.google.com/accounts/Login?continue=http%3A//localhost/signup%3Freturn_url%3Dhttp%253A%252F%252Flocalhost%252Fadmin'**

Reply:
as I have said earlier you'll get redirected to the login page if you'll access the admin tab with a guest user! Try doing this on the local server!

Reply:
Done!

Reply:
Hello @seanlip, would you review my PR